WalletWasabi/Wallets/P2pBlockProvider.cs

```diff
@@ -109,14 +109,14 @@ public async Task<Block> GetBlockAsync(uint256 hash, CancellationToken cancellat
 							if (!block.Check())
 							{
 								Logger.LogInfo($"Disconnected node: {node.RemoteSocketAddress}, because invalid block received.");
-								node.DisconnectAsync("Invalid block received.");
+								DisconnectNode(node, "Invalid block received.", force: true);
 								continue;
 							}
 
 							if (Nodes.ConnectedNodes.Count > 1) // To minimize risking missing unconfirmed transactions.
 							{
 								Logger.LogInfo($"Disconnected node: {node.RemoteSocketAddress}. Block downloaded: {block.GetHash()}.");
-								node.DisconnectAsync("Thank you!");
+								DisconnectNode(node, "Thank you!");
```

Comment:
Just curious - do we send `Thank you!` into the logs of the peer that has given us the block? If yes - that's fun!

Reply:
We log that we disconnected the node in the line above, but this might be not true.

Reply:
No @MaxHillebrand, we just write `Thank you` in a local log.